### PR TITLE
fetch and save client

### PR DESCRIPTION
**Pull request by edewit**

### CODE REVIEW

File: src/clients/messages.json [hunk 1]
```diff
@@ -14,6 +14,8 @@
     "capabilityConfig": "Capability config",
     "clientsExplain": "Clients are applications and services that can request authentication of a user",
     "clientImportError": "Could not import client",
+    "clientSaveSuccess": "Client successful updated",
```
Comment (ssilvert) [inline src/clients/messages.json:17]: Should say either `Client successfully updated` or `Client update successful`

File: src/clients/messages.json [hunk 2]
```diff
@@ -14,6 +14,8 @@
...
+    "clientSaveError": "Client could not be updated:",
     "clientImportSuccess": "Client imported successful",
```
Comment (ssilvert) [inline src/clients/messages.json:19]: Should say either `Client import successful` or `Client imported successfully`

File: src/clients/ClientSettings.tsx [hunk 3]
```diff
@@ -1,45 +1,73 @@
...
-  client: ClientRepresentation;
-};
-
-export const ClientSettings = ({ client: clientInit }: ClientSettingsProps) => {
+export const ClientSettings = () => {
   const { t } = useTranslation("clients");
-  const [client, setClient] = useState({ ...clientInit });
+  const httpClient = useContext(HttpClientContext)!;
+  const { realm } = useContext(RealmContext);
+  const [add, Alerts] = useAlerts();
```
Comment (ssilvert) [inline src/clients/ClientSettings.tsx:34]: Can we change this from `add` to `alert`?   `add` is too generic.  The code will read much nicer if we change it.

**Review by ssilvert — CHANGES_REQUESTED** [review]

File: src/clients/ClientList.tsx [hunk 4]
```diff
@@ -36,16 +36,13 @@ export const ClientList = ({ baseUrl, clients }: ClientListProps) => {
...
-    return field.indexOf("true") !== -1 ? (
-      <Link to="client-settings">{value}</Link>
-    ) : (
-      <Link to="client-settings">
-        {value} <Badge isRead>Disabled</Badge>
+    const [id, clientId] = field.split("#");
+    return (
+      <Link to={`client-settings/${id}`}>
+        {clientId}
+        {field.indexOf("true") === -1 && <Badge isRead>Disabled</Badge>}
```
Comment (dlabaj) [inline src/clients/ClientList.tsx:45]: Can we use a constant instead of -1

File: src/clients/ClientList.tsx [hunk 5]
```diff
@@ -76,7 +73,7 @@ export const ClientList = ({ baseUrl, clients }: ClientListProps) => {
 
   const data = clients!
     .map((r) => {
-      r.clientId = r.clientId + "#" + r.enabled;
+      r.clientId = r.id + "#" + r.clientId + "#" + r.enabled;
```
Comment (dlabaj) [inline src/clients/ClientList.tsx:76]: Use something other then r to describe the argument.

File: src/clients/ClientList.tsx [hunk 6]
```diff
@@ -103,7 +100,7 @@ export const ClientList = ({ baseUrl, clients }: ClientListProps) => {
             title: t("common:export"),
             onClick: (_, rowId) => {
               const clientCopy = JSON.parse(JSON.stringify(data[rowId].client));
-              clientCopy.clientId = convertClientId(clientCopy.clientId);
+              clientCopy.clientId = clientCopy.clientId.split("#")[1];
```
Comment (dlabaj) [inline src/clients/ClientList.tsx:103]: Use a constant instead of 1

File: src/clients/ClientSettings.tsx [hunk 7]
```diff
@@ -1,45 +1,73 @@
...
-    const target = event.target;
-    const name = (target as HTMLInputElement).name;
+  const url = `/admin/realms/${realm}/clients/${id}`;
 
-    setClient({
-      ...client,
-      [name]: value,
-    });
+  useEffect(() => {
+    const func = async () => {
```
Comment (dlabaj) [inline src/clients/ClientSettings.tsx:41]: Can you do this with an IIFE instead of declaring a const?

File: src/clients/messages.json (hunk 1 above)
Comment (dlabaj) [reply src/clients/messages.json:17]: Like Client successfully updated.

File: src/components/multi-line-input/MultiLineInput.tsx [hunk 8]
```diff
@@ -9,6 +9,20 @@ import {
 } from "@patternfly/react-core";
 import { MinusIcon, PlusIcon } from "@patternfly/react-icons";
 
+type MultiLine = {
+  value: string;
+};
+
+export function convertToMultiline(field: string[]): MultiLine[] {
+  return field.map((f) => {
```
Comment (dlabaj) [inline src/components/multi-line-input/MultiLineInput.tsx:17]: change argument to something other then f.

**Review by dlabaj — COMMENTED** [review]
Looks good a few minor comments.

File: src/clients/ClientList.tsx (hunk 4 above)
Comment (edewit) [reply src/clients/ClientList.tsx:45]: I think usually people don't create constants for -1

Comment (dlabaj) [reply src/clients/ClientList.tsx]: okay

**Review by dlabaj — APPROVED** [review]

**Review by ssilvert — APPROVED** [review]

**Pull request merged**